checks.d/yarn.py

```diff
@@ -196,9 +197,22 @@ def _yarn_app_metrics(self, rm_address, addl_tags):
 
                     for app_json in metrics_json['apps']['app']:
 
-                        app_name = app_json['name']
+                        tags = []
+
+                        for tag in app_tags:
+
+                            try:
+                                key, value = tag.split(':')
+                            except ValueError:
+                                self.log.error("Invalid value {0} for application_tag").format()
+
+                            try:
+                                tags.append("{tag}:{value}".format(
```

Comment:
This will fail if the previous block failed (if `ValueError` was raised).

I think it would be better to have something like:
```
try:
    key, value = tag.split(':')
    tags.append(...)
except ValueError:
    ...
except KeyError:
    ...
```

This way is the first fails, the second is not executed.